zerver/models.py

```diff
@@ -3111,20 +3137,26 @@ def sent_by_human(self) -> bool:
         sending_client = self.sending_client.name.lower()
 
         return (
-            sending_client
-            in (
-                "zulipandroid",
-                "zulipios",
-                "zulipdesktop",
-                "zulipmobile",
-                "zulipelectron",
-                "zulipterminal",
-                "snipe",
-                "website",
-                "ios",
-                "android",
+            (
+                sending_client
+                in (
+                    "zulipandroid",
+                    "zulipios",
+                    "zulipdesktop",
+                    "zulipmobile",
+                    "zulipelectron",
+                    "zulipterminal",
+                    "snipe",
+                    "website",
+                    "ios",
+                    "android",
+                )
             )
-        ) or ("desktop app" in sending_client)
+            or ("desktop app" in sending_client)
+            # Since the vast majority of messages are sent by humans
+            # in Zulip, treat test suite messages as such.
+            or (sending_client == "test suite" and settings.TEST_SUITE)
+        )
```

Comment:
Applied this revision so that this logic would only apply in the test suite; it's best to avoid changing production outcomes for testing reasons even if it's probably not going to come up in practice:

```diff
diff --git a/zerver/models.py b/zerver/models.py
index 55e958d22f..10129cb14d 100644
--- a/zerver/models.py
+++ b/zerver/models.py
@@ -3137,21 +3137,26 @@ class Message(AbstractMessage):
         sending_client = self.sending_client.name.lower()
 
         return (
-            sending_client
-            in (
-                "zulipandroid",
-                "zulipios",
-                "zulipdesktop",
-                "zulipmobile",
-                "zulipelectron",
-                "zulipterminal",
-                "snipe",
-                "website",
-                "ios",
-                "android",
-                "test suite",
+            (
+                sending_client
+                in (
+                    "zulipandroid",
+                    "zulipios",
+                    "zulipdesktop",
+                    "zulipmobile",
+                    "zulipelectron",
+                    "zulipterminal",
+                    "snipe",
+                    "website",
+                    "ios",
+                    "android",
+                )
             )
-        ) or ("desktop app" in sending_client)
+            or ("desktop app" in sending_client)
+            # Since the vast majority of messages are sent by humans
+            # in Zulip, treat test suite messages as such.
+            or (sending_client == "test suite" and settings.TEST_SUITE)
+        )
 
     @staticmethod
     def is_status_message(content: str, rendered_content: str) -> bool:
```